File: examples/basic/src/main/java/de/whitefrog/frogr/example/rest/Persons.java
package de.whitefrog.frogr.example.rest;

import de.whitefrog.frogr.example.model.Person;
import de.whitefrog.frogr.example.repository.PersonRepository;
import de.whitefrog.frogr.model.SearchParameter;
import de.whitefrog.frogr.rest.request.SearchParam;
import de.whitefrog.frogr.rest.service.CRUDService;
import org.neo4j.graphdb.Transaction;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import java.util.List;

@Path("person")
public class Persons extends CRUDService<PersonRepository, Person> {
  @GET
  @Path("init")
  public void init() {
    // insert some data
    try(Transaction tx = service().beginTx()) {
      if(repository().search().count() == 0) {
        repository().init();
        tx.success();
      }
    }
  }
  
  @GET
  @Path("custom-search")
  public List<Person> customSearch(@SearchParam SearchParameter params) {
    try(Transaction ignored = service().beginTx()) {
      return repository().search().params(params).list();
    }
  }
}

File: base/src/test/java/de/whitefrog/frogr/test/rest/Persons.kt
package de.whitefrog.frogr.test.rest

import de.whitefrog.frogr.model.SearchParameter
import de.whitefrog.frogr.rest.request.SearchParam
import de.whitefrog.frogr.rest.service.CRUDService
import de.whitefrog.frogr.test.model.Person
import de.whitefrog.frogr.test.repository.PersonRepository
import javax.ws.rs.GET
import javax.ws.rs.Path

@Path("person")
class Persons: CRUDService<PersonRepository, Person>()